lib/dialects/postgres/query-generator.js

```diff
@@ -720,36 +720,50 @@ var QueryGenerator = {
     }).join(' OR ');
   },
 
-  pgEnumName: function (tableName, attr, options) {
+  pgExtractTableDetails: function(tableName, options) {
     options = options || {};
+    tableName = tableName || {};
+    var tableDetails = {
+      schema: tableName.schema || options.schema || 'public',
+      tableName: _.isPlainObject(tableName) ? tableName.table : tableName,
+      delimiter: tableName.delimiter || options.delimiter || '.'
+    };
+
+    return tableDetails;
+  },
```

Comment:
Done. When you've approved this PR in principle I will write some tests for extractTableDetails